pkg/controllers/state/metrics.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	crmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	"sigs.k8s.io/karpenter/pkg/metrics"
+)
+
+const (
+	stateSubsystem = "state"
+	SyncedKey      = "synced"
+)
+
+var (
+	ClusterStateNodesTotal = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: metrics.Namespace,
+			Subsystem: stateSubsystem,
+			Name:      "nodes_total",
+			Help:      "total nodes in karpenter's cluster state",
+		},
+		[]string{},
+	)
+
+	ClusterStateSynced = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: metrics.Namespace,
+			Subsystem: stateSubsystem,
+			Name:      "synced",
+			Help:      "synced is 1 if the nodeclaims and nodes stored in the apiserver have the same representation in cluster state",
+		},
+		[]string{SyncedKey},
```

Comment:
Alternatively we could not include the SyncedKey since the 0, 1 indicates clearly if the cluster is synced

Reply:
It's less about cardinality, more about unnecessary redundancy. I think having 1/0 is sufficient here. Do you see any additional value to having the label as well?

Reply:
Removed it